### ISSUE TITLE
Client not disconnects from HZ server

### ISSUE THREAD

**Issue by dsukhoroslov** [hazelcast/hazelcast#9402]
I have a HZ server and Java client connected to it. After performing several operations (just 1 is enough) I disconnect client doing:
    hzClient.getLifecycleService().shutdown();
or (also tried)
    hzClient.shutdown();

the log output is:
```
2016-12-08 16:59:11.439 [Thread-1] INFO  com.hazelcast.core.LifecycleService - hz.client_0 [default] [3.7.3] HazelcastClient 3.7.3 (20161117 - 8166eea) is SHUTTING_DOWN
2016-12-08 16:59:11.447 [Thread-1] INFO  com.hazelcast.core.LifecycleService - hz.client_0 [default] [3.7.3] HazelcastClient 3.7.3 (20161117 - 8166eea) is SHUTDOWN
```
but after that the client app not terminates, but still waiting for something. In JMX console I see the hz.client_0.event-1 thread in Park state.

This happens in HZ 3.7.3 only. In HZ 3.7.2 the same scenario work fine: client app terminates after client disconnection.

**Comment by jerrinot** [comment]
hi @dsukhoroslov,

this sounds like an issue fixed by https://github.com/hazelcast/hazelcast/pull/9392
The fix is in the 3.7.4 release will be released in upcoming days.

**Comment by dsukhoroslov** [comment]
yes, looks like this is the cause. Ok, will wait for 3.7.4, thank you @jerrinot !
